capa/features/extractors/elffile.py

```diff
@@ -13,19 +13,36 @@
 from elftools.elf.elffile import ELFFile, SymbolTableSection
 
 import capa.features.extractors.common
-from capa.features.file import Import, Section
+from capa.features.file import Export, Import, Section
 from capa.features.common import OS, FORMAT_ELF, Arch, Format, Feature
 from capa.features.address import NO_ADDRESS, FileOffsetAddress, AbsoluteVirtualAddress
 from capa.features.extractors.base_extractor import FeatureExtractor
 
 logger = logging.getLogger(__name__)
 
 
-def extract_file_import_names(elf, **kwargs):
-    # see https://github.com/eliben/pyelftools/blob/0664de05ed2db3d39041e2d51d19622a8ef4fb0f/scripts/readelf.py#L372
-    symbol_tables = [(idx, s) for idx, s in enumerate(elf.iter_sections()) if isinstance(s, SymbolTableSection)]
+def extract_file_export_names(elf, **kwargs):
+    for _, section in enumerate(elf.iter_sections()):
+        if not isinstance(section, SymbolTableSection):
+            continue
+
+        if section["sh_entsize"] == 0:
+            logger.debug("Symbol table '%s' has a sh_entsize of zero!", section.name)
+            continue
 
-    for _, section in symbol_tables:
+        logger.debug("Symbol table '%s' contains %s entries:", section.name, section.num_symbols())
+
+        for _, symbol in enumerate(section.iter_symbols()):
+            # The following conditions are based on the following article
+            # http://www.m4b.io/elf/export/binary/analysis/2015/05/25/what-is-an-elf-export.html
+            if symbol.name and symbol.entry.st_info.type in ["STT_FUNC", "STT_OBJECT", "STT_IFUNC"]:
+                if symbol.entry.st_value != 0 and symbol.entry.st_shndx != "SHN_UNDEF":
+                    # Export symbol
+                    yield Export(symbol.name), AbsoluteVirtualAddress(symbol.entry.st_value)
```

Comment:
```suggestion
            if not symbol.name:
                continue
            if symbol.entry.st_info.type not in ["STT_FUNC", "STT_OBJECT", "STT_IFUNC"]:
                continue
            if symbol.entry.st_value == 0:
                continue
            if symbol.entry.st_shndx == "SHN_UNDEF":
                continue

            # Export symbol
            yield Export(symbol.name), AbsoluteVirtualAddress(symbol.entry.st_value)
```

Reply:
```
Container({'st_name': 489, 'st_info': Container({'bind': 'STB_GLOBAL', 'type': 'STT_OBJECT'}), 'st_shndx': 25, 'st_value': 16400, 'st_size': 0})
Container({'st_name': 501, 'st_info': Container({'bind': 'STB_WEAK', 'type': 'STT_NOTYPE'}), 'st_shndx': 'SHN_UNDEF', 'st_value': 0, 'st_size': 0})
```
In above examples of `symbol.entry` import st_shndx = 'SHN_UNDEF' ( alias for **0** ) and in case of export a number.